dashboard/app/models/pd/scholarship_info.rb

```diff
@@ -0,0 +1,38 @@
+# == Schema Information
+#
+# Table name: pd_scholarship_infos
+#
+#  id                 :integer          not null, primary key
+#  user_id            :integer          not null
+#  application_year   :string(255)      not null
+#  scholarship_status :string(255)      not null
+#  pd_application_id  :integer
+#  pd_enrollment_id   :integer
+#  created_at         :datetime         not null
+#  updated_at         :datetime         not null
+#
+# Indexes
+#
+#  index_pd_scholarship_infos_on_pd_application_id  (pd_application_id)
+#  index_pd_scholarship_infos_on_pd_enrollment_id   (pd_enrollment_id)
+#  index_pd_scholarship_infos_on_user_id            (user_id)
+#
+
+class Pd::ScholarshipInfo < ActiveRecord::Base
+  include Pd::Application::ActiveApplicationModels
+  include Pd::Application::ApplicationConstants
+  include Pd::ScholarshipInfoConstants
+
+  # We began using scholarships in 2019-2020, so remove 2018-2019 from this list
+  SCHOLARSHIP_YEARS = APPLICATION_YEARS.drop(1).freeze
+
+  belongs_to :user
+  belongs_to :enrollment, class_name: 'Pd::Enrollment', foreign_key: :pd_enrollment_id
+  belongs_to :application, class_name: 'Pd::Application::ApplicationBase', foreign_key: :pd_application_id
```

Comment:
Question (I genuinely don't know): Does using `ApplicationBase` as the class_name here mess with the STI pattern we're using for applications at all?

Reply:
I could use `TeacherApplicationBase` instead - that might be helpful in communicating that only teacher applications have scholarship info. In my testing, I didn't find any issues with this - when you call `scholarship_info.pd_application.class` it returns `Pd::Application::Teacher1920Application`

Reply:
The benefit of using `TeacherApplicationBase` as the class name is that when setting `scholarship_info.application = <PdApplication>` you will get an error if you try to assign a non-teacher application. The flipside is that if someone mistakenly creates a scholarship info object for a different kind of application (which you could do in the db since the different application types live in the same table), then calling `scholarship_info.application` on it will return `nil`, even though technically there is an associated application.

Reply:
But I think that is better than leaving it as ApplicationBase, which doesn't give you an error for trying to assign the scholarship info to a non-teacher application.